src/NHibernate/Driver/OracleDataClientDriverBase.cs

```diff
@@ -0,0 +1,176 @@
+
+		private const string _unmanagedProviderInvariantName = "Oracle.DataAccess.Client";
+		private const string _unmanagedDriverAssemblyName = "Oracle.DataAccess";
+		private const string _unmanagedClientNamespace = "Oracle.DataAccess.Client";
+
+		private const string _managedProviderInvariantName = "Oracle.ManagedDataAccess.Client";
+		private const string _managedDriverAssemblyName = "Oracle.ManagedDataAccess";
+		private const string _managedClientNamespace = "Oracle.ManagedDataAccess.Client";
+
+		private static readonly SqlType _guidSqlType = new SqlType(DbType.Binary, 16);
+		private readonly PropertyInfo _oracleCommandBindByName;
+		private readonly PropertyInfo _oracleDbType;
+		private readonly object _oracleDbTypeRefCursor;
+		private readonly object _oracleDbTypeXmlType;
+		private readonly object _oracleDbTypeNVarchar2;
+		private readonly object _oracleDbTypeNChar;
+
+		/// <summary>
+		/// Default constructor.
+		/// </summary>
+		/// <param name="managed"><see langword="true"/> for loading the managed driver, <see langword="false"/> for loading the unmanaged driver.</param>
+		/// <exception cref="HibernateException">
+		/// Thrown when the <c>Oracle.DataAccess</c> assembly can not be loaded.
+		/// </exception>
+		protected OracleDataClientDriverBase(bool managed)
+			: this(
+				managed ? _managedProviderInvariantName : _unmanagedProviderInvariantName,
+				managed ? _managedDriverAssemblyName : _unmanagedDriverAssemblyName,
+				managed ? _managedClientNamespace : _unmanagedClientNamespace)
+		{
+		}
+
+		private OracleDataClientDriverBase(string providerInvariantName, string driverAssemblyName, string clientNamespace)
+			: base(providerInvariantName, driverAssemblyName, clientNamespace + ".OracleConnection", clientNamespace + "." + _commandClassName)
+		{
+			var oracleCommandType = ReflectHelper.TypeFromAssembly(clientNamespace + "." + _commandClassName, driverAssemblyName, true);
+			_oracleCommandBindByName = oracleCommandType.GetProperty("BindByName");
+
+			var parameterType = ReflectHelper.TypeFromAssembly(clientNamespace + ".OracleParameter", driverAssemblyName, true);
+			_oracleDbType = parameterType.GetProperty("OracleDbType");
+
+			var oracleDbTypeEnum = ReflectHelper.TypeFromAssembly(clientNamespace + ".OracleDbType", driverAssemblyName, true);
+			_oracleDbTypeRefCursor = Enum.Parse(oracleDbTypeEnum, "RefCursor");
+			_oracleDbTypeXmlType = Enum.Parse(oracleDbTypeEnum, "XmlType");
+			_oracleDbTypeNVarchar2 = Enum.Parse(oracleDbTypeEnum, "NVarchar2");
+			_oracleDbTypeNChar = Enum.Parse(oracleDbTypeEnum, "NChar");
+		}
+
+		/// <inheritdoc/>
+		public override void Configure(IDictionary<string, string> settings)
+		{
+			base.Configure(settings);
+
+			// If changing the default value, keep it in sync with Oracle8iDialect.Configure.
+			UseNPrefixedTypesForUnicode = PropertiesHelper.GetBoolean(Cfg.Environment.OracleUseNPrefixedTypesForUnicode, settings, false);
```

Comment:
Default value `false` avoid a possible breaking change for queries, by keeping the default typing of the ODP.Net provider.